commands/validators.js

```diff
@@ -0,0 +1,114 @@
+const exitOnError = require('../utils/exit-on-error');
+const connect = require('../utils/connect');
+const { utils } = require('near-api-js');
+const BN = require('bn.js');
+const AsciiTable = require('ascii-table');
+
+function findSeatPrice(validators, numSeats) {
+    let stakes = validators.map((v) => new BN(v.stake, 10)).sort((a, b) => a.cmp(b));
+    let num = new BN(numSeats);
+    let stakesSum = stakes.reduce((a, b) => a.add(b));
+    // assert stakesSum >= numSeats
+    let left = new BN(1), right = stakesSum.add(new BN(1));
+    while (!left.eq(right.sub(new BN(1)))) {
+        const mid = left.add(right).div(new BN(2));
+        let found = false;
+        let currentSum = new BN(0);
+        for (let i = 0; i < stakes.length; ++i) {
+            currentSum = currentSum.add(stakes[i].div(mid));
+            if (currentSum.gte(num)) {
+                left = mid;
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            right = mid;
+        }
+    }
+    return left;
+}
+
+function diffValidators(currentValidators, nextValidators) {
+    const result = { newValidators: [], removedValidators: [], changeValidators: [] };
+    const validatorsMap = new Map();
+    const nextValidatorsMap = new Map();
+    nextValidators.map((validator) => nextValidatorsMap[validator.account_id] = validator);
+    currentValidators.map((validator) => { 
+        if (!(validator.account_id in nextValidatorsMap)) {
+            result.removedValidators.push(validator);
+        }
+        validatorsMap[validator.account_id] = validator;
+    });
+    nextValidators.map((validator) => {
+        if (validator.account_id in validatorsMap) {
+            if (validatorsMap[validator.account_id].stake != validator.stake) {
+                result.changeValidators.push({new: validator, old: validatorsMap[validator.account_id]});
+            }
+        } else {
+            result.newValidators.push(validator);
+        }
+    });
+    return result;
+}
+
+module.exports = {
+    command: 'EXPERIMENTAL_validators',
+    desc: 'lookup validators and proposals',
+    handler: exitOnError(async (argv) => {
```

Comment:
same as with `findSeatPrice` looks like too much logic lives in `near-shell` instead of `near-api-js`. Like if I was building web app to do the same thing, I'll probably need the same logic minus the `AsciiTable`.